src/RouteUsage.php

```diff
@@ -3,9 +3,12 @@
 namespace Julienbourdeau\RouteUsage;
 
 use Illuminate\Database\Eloquent\Model;
+use Julienbourdeau\RouteUsage\Traits\AuthorizesRequests;
 
 class RouteUsage extends Model
 {
+    use AuthorizesRequests;
```

Comment:
Did you mean to put this on the controller?

Reply:
Hi @julienbourdeau this is the same with Laravel Telescope `AuthorizesRequests` trait

https://github.com/laravel/telescope/blob/2.0/src/AuthorizesRequests.php

Where you can override on how you can authorize users from accessing your `/route-usage` path.

```php
// You can add this to your AppServiceProvider
RouteUsage::auth(function ($request) {
    return $request->user()->is_admin;
});
```

https://github.com/laravel/telescope/blob/2.0/src/TelescopeApplicationServiceProvider.php#L29

Reply:
And because of the above code user can have a greater control on accessing the dashboard route and can declare middleware

Reply:
Both Horizon and Telescope add this behavior to a general config class, here `RouteUsage` is a model. I'd definitely prefer not having auth concern in the model.

I think a middleware and a Gate is enough: https://github.com/julienbourdeau/route-usage/pull/13/files#diff-c9b107c3af2c5c8ed5663462cb5c668bR16